Detections/Dynamics365Activity/MassExportOfDynamicstoExcel.yaml

```diff
@@ -0,0 +1,52 @@
+id: 05eca115-c4b5-48e4-ba6e-07db57695be2
+name: Mass Export of Dynamics 365 Records to Excel
+description: |
+  'The query detects user exporting a large amount of records from Dynamics 365 to Excel, significantly more records exported than any other recent activity by that user.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Dynamics365
+    dataTypes:
+      - Dynamics365Activity
+queryFrequency: 1d
+queryPeriod: 7d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Collection
+relevantTechniques:
+  - T1530
+query: |
+  let baseline_time = 7d;
+  let detection_time = 1d;
+  Dynamics365Activity
+  | where TimeGenerated between(ago(baseline_time)..ago(detection_time-1d))
+  | where OriginalObjectId contains 'ExportToExcel'
+  | extend numQueryCount = todouble(QueryResults)
+  | extend QueryCount = iif(QueryResults contains ",", todouble(countof(tostring(QueryResults), ',') + 1), numQueryCount)
```

Comment:
I don't have a test environment with this, but would it have better perf if we use array_length here?  Looks like QueryResult is already a dynamic?

Reply:
It is a dynamic but not in that way. The type depends on the size of query, sometimes its a int, sometimes a double, and sometimes a string of ids. This query is very clunkly but unfortunately its needed due to the data.